modules/apps/portal-search-opensearch2/portal-search-opensearch2-impl/src/main/java/com/liferay/portal/search/opensearch2/internal/document/OpenSearchDocumentFactoryImpl.java

```diff
@@ -42,11 +43,11 @@
 public class OpenSearchDocumentFactoryImpl
 	implements OpenSearchDocumentFactory {
 
-	public static final FastDateFormat DATE_FORMAT = FastDateFormat.getInstance(
-		"yyyyMMddHHmmss");
-
 	public static final String DATE_MAX_VALUE = "99950812133000";
 
+	public static final Format FORMAT =
+		FastDateFormatFactoryUtil.getSimpleDateFormat("yyyyMMddHHmmss");
+
```

Comment:
@julschong This `FastDateFormatFactoryUtil.getSimpleDateFormat("yyyyMMddHHmmss")` does not equal to `FastDateFormat.getInstance("yyyyMMddHHmmss")`.

Please figure out why and fix it. 

Tina.

Reply:
@tinatian I looked through commons-lang `FastDateFormat`

Orignal:
```
FastDateFormat.getSimpleDateFormat("yyyyMMddHHmmss")
v
FastDateFormat emptyFormat = new FastDateFormat(pattern("yyyMMddHHmmss"), timeZone(null), locale(null));
v
protected FastDateFormat(String pattern, TimeZone timeZone, Locale locale) {
        if (pattern == null) {
            throw new IllegalArgumentException("The pattern must not be null");
        } else {
            this.mPattern = pattern;
            this.mTimeZoneForced = timeZone != null;
            if (timeZone == null) {
                timeZone = TimeZone.getDefault(); <-----------------------------------
            }

            this.mTimeZone = timeZone;
            this.mLocaleForced = locale != null;
            if (locale == null) {
                locale = Locale.getDefault();  <------------------------------------
            }

            this.mLocale = locale;
        }
    }
```
with Util, this becomes `FastDateFormat.getInstance(pattern, timeZone(null), locale (Locale.getDefault()));`

The difference is on the locale passed into FastDateFormat.getInstance method.
This would affect the mLocaleForced variable in there
and mLocaleForced affects the result from hashcode() and equals method.

I can change it to use the long form Util method `FastDateFormatFactoryUtil.getSimpleDateFormat("yyyyMMddHHmmss", null, null) `